### ISSUE TITLE
uploadKey() - no file object

### ISSUE THREAD

**Issue by red010b37** [TTLabs/EvaporateJS#174]
Hi so I'm new to evaporator js and don't want to charge around to much as I might be missing a lot. 

But I grabbed the latest version and their seems to be an issue with the add and the "uploadKey()" function.

```
function uploadKey(fileUpload) {
            // The key tries to give a signature to a file in the absence of its path.
            // "<filename>-<mimetype>-<modifieddate>-<filesize>"
            return [
                fileUpload.file.name,
                fileUpload.file.type,
                dateISOString(fileUpload.file.lastModifiedDate),
                fileUpload.file.size
            ].join("-");
        }

```

I debugged the code and Its looking for the fileUpload.file object but it doesn't exist. However they seem to be top level properties - is this correct?

I think it should be 
```
function uploadKey(fileUpload) {
            // The key tries to give a signature to a file in the absence of its path.
            // "<filename>-<mimetype>-<modifieddate>-<filesize>"
            return [
                fileUpload.name,
                fileUpload.type,
                dateISOString(fileUpload.lastModifiedDate),
                fileUpload.size
            ].join("-");
        }

```

**Comment by red010b37** [comment]
actually it seems to be everywhere - its like the file obj is never handed over.

**Comment by red010b37** [comment]
I'm also using the aws4 signing example
```
<script language="javascript">
    var files;
    var _e_ = new Evaporate({
      signerUrl: 'http://localhost:8080/sign_auth',
      aws_key: 'xxxx',
      awsRegion: 'ap-southeast-2',
      aws_url: 'https://s3-ap-southeast-2.amazonaws.com',
      bucket: 'uploads',
      awsSignatureVersion: '4',
      computeContentMd5: true,
      cryptoMd5Method: function (data) { return AWS.util.crypto.md5(data, 'base64'); },
      cryptoHexEncodedHash256: function (data) { return AWS.util.crypto.sha256(data, 'hex'); }
    });
    $('#files').change(function (evt) {
      files = evt.target.files;
      for (var i = 0; i < files.length; i++) {
        _e_.add(files[i], {
          name: 'test_' + Math.floor(1000000000*Math.random()),
          complete: function () {
            console.log('complete................yay!');
          },
          progress: function (progress) {
            console.log('making progress: ' + progress);
          }
        });
      }
      $(evt.target).val('');
    });

  </script>
```

**Comment by red010b37** [comment]
Any help would be much appreciated.

**Comment by njbmartin** [comment]
What browser are you using? fileUpload`.file` is part of the File API, and thus is required for the upload to work.

You can check that it should be supported with `_e_.supported`. Also, check your console outputs for any errors.

**Comment by njbmartin** [comment]
Ignore that, I've just updated to the latest version and ran into issues myself.

Evaporate is a little confused about how it handles files and backwards compatibility is also partially broken (see PR #175), but @bikeath1337 is already looking into it (PR #173)

**Comment by red010b37** [comment]
@njbmartin thanks for confirming that. Do you know which is the last release that was functioning correctly?

**Comment by bikeath1337** [comment]
I will look at this in the morning. I have an outstanding PR that I'll put in. Apologies for the delay.

**Comment by bikeath1337** [comment]
1.4.1 should be working fine.

**Comment by red010b37** [comment]
@bikeath1337  great thanks :)

**Comment by bikeath1337** [comment]
I restored the documentation and behavior that was breaking this and published it as v1.4.4.

**Comment by bikeath1337** [comment]
@red010b37 did this fix solve your issue?
